docs/template_plugin/tests/functional/op_reference/if.cpp

```diff
@@ -0,0 +1,302 @@
+        auto Xe = std::make_shared<op::Parameter>(element::f32, PartialShape::dynamic());
+        auto Ye = std::make_shared<op::Parameter>(element::f32, PartialShape::dynamic());
+        // Body
+        auto then_op = std::make_shared<op::v1::Multiply>(Xt, Yt);
+        auto else_op = std::make_shared<op::v1::Add>(Xe, Ye);
+        auto then_op_result = std::make_shared<op::Result>(then_op);
+        auto else_op_result = std::make_shared<op::Result>(else_op);
+        auto then_body = std::make_shared<ngraph::Function>(OutputVector{then_op_result}, ParameterVector{Xt, Yt});
+        auto else_body = std::make_shared<ngraph::Function>(OutputVector{else_op_result}, ParameterVector{Xe, Ye});
+        auto if_op = std::make_shared<op::v8::If>(cond);
+        if_op->set_then_body(then_body);
+        if_op->set_else_body(else_body);
+        if_op->set_input(X, Xt, Xe);
+        if_op->set_input(Y, Yt, Ye);
+        auto res = if_op->set_output(then_op_result, else_op_result);
+        if_op->validate_and_infer_types();
+        std::vector<float> X_v{1.0, 2.0, 3.0, 4.0};
+        std::vector<float> Y_v{2.0, 1.0, 2.0, 3.0};
+        auto fun = std::make_shared<Function>(OutputVector{res}, ParameterVector{cond, X, Y});
+        return fun;
+    }
+};
+
+struct IfParams {
+    IfParams(const std::shared_ptr<IfFunctionalBase>& functional,
+             const std::vector<Tensor>& if_inputs,
+             const std::vector<Tensor>& expected_results,
+             const std::string& test_case_name)
+        : function(functional),
+          inputs(if_inputs),
+          expected_results(expected_results),
+          test_case_name(test_case_name) {}
+
+    std::shared_ptr<IfFunctionalBase> function;
+    std::vector<Tensor> inputs;
+    std::vector<Tensor> expected_results;
+    std::string test_case_name;
+};
+
+class ReferenceIfLayerTest : public testing::TestWithParam<IfParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = params.function->create_function(params.inputs, params.expected_results);
+        inputData.reserve(params.inputs.size());
+        refOutData.reserve(params.expected_results.size());
+        for (auto& input_tensor : params.inputs) {
+            inputData.push_back(input_tensor.data);
+        }
+        for (auto& expected_tensor : params.expected_results) {
+            refOutData.push_back(expected_tensor.data);
+        }
+    }
+    static std::string getTestCaseName(const testing::TestParamInfo<IfParams>& obj) {
+        auto param = obj.param;
+        return param.test_case_name;
+    }
+};
+
+std::vector<float> y_gen() {
+    std::vector<float> result;
+    for (auto c_ind = 0; c_ind < 4; ++c_ind) {
+        for (auto d_ind = 0; d_ind < 4; ++d_ind) {
+            result.push_back(static_cast<float>(c_ind * d_ind));
+        }
+    }
+    return result;
+}
+
+std::vector<float> z_gen() {
```

Comment:
Where do you use functions `y_gen()` and `z_gen()`? If these functions are used, can these functions be `static` or members of anonymous namespace (i.e. `namespace {...}`)?